### DISCUSSION TITLE
specifying egress port

### DISCUSSION THREAD

**Discussion by fernandohv3279** [microsoft/msquic#2813] [Q&A] [answered]
Hello, I would like to know if there is a way to specify the Egress port on the client side on the sample.c code. In sample.c the server listens on port 4567, the client sends data through a random port, how can I specify that port?

**Comment by nibanks** [comment, marked as answer]
You'll want to set (via `SetParam`) `QUIC_PARAM_CONN_LOCAL_ADDRESS` to set the local address before starting the connection.

**Reply by fernandohv3279** [reply]
Thank you. I got it working by adding these 3 lines before ConnectionStart:

```
QUIC_ADDR addr;
QuicAddrFromString("127.0.0.1", 9999, &addr);
MsQuic->SetParam(Connection, QUIC_PARAM_CONN_LOCAL_ADDRESS, sizeof(addr), &addr);

```
